my-app/src/App/pages/MarketPage/components/ButtonSerach/ButtonSearch.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import styles from "./ButtonSearch.module.scss";
+// import {Loader, LoaderSize} from "../Loader/Loader";
+
+export type ButtonProps = React.PropsWithChildren<{
+  loading?: boolean;
+}> &
+  React.ButtonHTMLAttributes<HTMLButtonElement>;
+
+export const ButtonSearch: React.FC<ButtonProps> = ({
+  onClick,
+  children,
+  loading = false,
+  className,
+  ...props
+}) => {
+  return (
+    <button
+      {...props}
+      onClick={loading !== true ? onClick : undefined}
+      className={styles.button}
```

Comment:
нужно использовать утилиту `classnames` https://www.npmjs.com/package/classnames и сделать вот так
```ts
import cn from 'classnames';
```
```suggestion
      className={cn(styles.button, className)}
```
чтобы можно было прокинуть класс снаружи